internal/api/v1/admin/accesskeys/admin_access_key_filter_query_builder.go

```diff
@@ -0,0 +1,28 @@
+package accesskeys
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/querybuilders"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/user/accesskeys"
+	"github.com/bitcoin-sv/spv-wallet/models/filter"
+)
+
+type adminAccessKeyFilterQueryBuilder struct {
+	adminAccessKeyFilter filter.AdminAccessKeyFilter
+}
+
+func (a *adminAccessKeyFilterQueryBuilder) Build() (url.Values, error) {
```

Comment:
what about adding validation to url ?
```suggestion
func (a *adminAccessKeyFilterQueryBuilder) Build() (url.Values, error) {
if err := a.validateFilter(); err != nil {
		return nil, err
	}
```
where 
```go
func (a *adminAccessKeyFilterQueryBuilder) validateFilter() error {
	if a.adminAccessKeyFilter.XpubID == "" {
		return fmt.Errorf("adminAccessKeyFilterQueryBuilder: xPubId is empty")
	}
	// Add more validation checks as needed
	return nil
}
```

Reply:
There’s no need for this validation because none of these parameters are required to construct the query. As a team, we agreed to use functional options to emphasize that query methods can work with or without parameters.

Even if someone sets filter values to zero, those fields will be skipped, and the resulting query won’t include any unexpected parameter chains. This behavior is handled internally by the ExtendedURLValues structure, which is consistently used across all query builders.

```
func (e *ExtendedURLValues) AddPair(key string, val any) {
	if val == nil || len(key) == 0 {
		return
	}
...
```

To better understand the workflow, I encourage you to run some runnable examples in debug mode—first with specific filters set, and then without them. This should help clarify how the process behaves in different scenarios. 😸